### PR TITLE
global: improvement of project structure

### PR DESCRIPTION
**Pull request by drjova**
* Improves project structure by splitting the components.

* Improves test feedback by adding test coverage report when you run
  ``npm test``.

Signed-off-by: Harris Tzovanakis <me@drjova.com>

### CODE REVIEW

File: src/invenio-files-js/templates/list.html [hunk 1]
```diff
@@ -1,3 +1,4 @@
+HELLO IS IT ME THAT LOOKING FOR
```
Comment (lnielsen) [inline src/invenio-files-js/templates/list.html]: Supposed to be here?

**Comment by lnielsen** [comment]
@drjova LGTM besides the little comment :-) Will you remove it and add a release commit at the same time so we can push it to npm

File: src/invenio-files-js/templates/list.html (hunk 1 above)
Comment (drjova) [inline src/invenio-files-js/templates/list.html]: 😶

**Comment by jirikuncar** [comment]
> release commit at the same time so we can push it to npm

What about following practice which allows you to re-push tag in case there is a problem with an automatic deployment.

1. create PR with release
2. wait for green light
3. push tag
4. wait for green light and check the release
5. merge it

**Comment by lnielsen** [comment]
@jirikuncar I see dangers with both approaches: pre/post-merge of release PR. Post-merge, you risk that you release something which accidentally we forget to merge to master (or alternatively someone else merges a PR in-between). Pre-merge you risk creating build which misses something, because you have to build it manually when the automatic deployment fails.

**Comment by jirikuncar** [comment]
@lnielsen I'm aware of pre-merge problem which is however minimal if we keep doing separate release PRs.

**Pull request merged**